source/D2Common/src/D2StatList.cpp

```diff
@@ -3471,79 +2275,42 @@ BOOL __stdcall D2Common_11274(D2UnitStrc* pTarget, D2UnitStrc* pUnit)
 
 	if (pUnit->pStatListEx->pUnit != pTarget)
 	{
-		STATLIST_MergeStatLists(pTarget, pUnit, 1);
+		STATLIST_MergeStatLists(pTarget, pUnit, addOrSubstract);
 		return FALSE;
 	}
 
-	if (!pTarget->pStatListEx || !(pTarget->pStatListEx->dwFlags & 0x80000000) || !(pUnit->pStatListEx->dwFlags & 0x40000000))
+	if (!(pTarget->pStatListEx && STATLIST_IsExtended(pTarget->pStatListEx) && (pUnit->pStatListEx->dwFlags & STATLIST_DYNAMIC)))
 	{
 		return FALSE;
 	}
 
-	pUnit->pStatListEx->dwFlags &= ~0x40000000;
+	pUnit->pStatListEx->dwFlags &= ~STATLIST_DYNAMIC;
```

Comment:
In 11275, it is pUnit->pStatListEx->dwFlags |= STATLIST_DYNAMIC;

The check above is also inverted